neqo-transport/src/connection.rs

```diff
@@ -2259,10 +2259,13 @@ impl Connection {
     }
 
     fn cleanup_streams(&mut self) {
+        self.send_streams.clear_terminal();
+
         let recv_to_remove = self
             .recv_streams
             .iter()
             .filter(|(_, stream)| stream.is_terminal())
+            .filter(|(id, _)| id.is_uni() || self.send_streams.get(**id).is_err())
             .map(|(id, _)| *id)
```

Comment:
Here is what I would do:

```rust
.filter_map(|(id, stream)| {
  // Remove all streams for which the receiving is done (or aborted).
  // But only if they are unidirectional, or we have finished sending.
  if stream.is_terminal() && (id.is_uni() || !self.send_streams.has_stream(**id)) {
      Some(*id)
  } else {
    None
  }
})
.collect::<Vec<_>>();
```

Reply:
Borrowck didn't like this, said "closure requires unique access to `self` but it is already borrowed", which I don't get -- send_streams and recv_streams should both be able to be borrowed at the same time. I think borrowck is being too conservative, honestly. So I'm going to leave it, but I like that comment and also adding a SendStreams method to test for existence.